component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordChangeHandler.java

```diff
@@ -89,6 +88,13 @@ public void handleEvent(Event event) throws IdentityEventException {
         }
 
         long timestamp = System.currentTimeMillis();
+        String tenantDomain = (String) event.getEventProperties()
+                .get(IdentityEventConstants.EventProperty.TENANT_DOMAIN);
+
+        // Store the last password update time in nanoseconds if the configuration is enabled.
+        if (isStoreTimeAsNanoseconds(tenantDomain)) {
```

Comment:
multiplying by 1000000 wont give the exact nanosecond time. It will only add zeros to the end of milisecond value. We need a way to get the correct nano second values. Can we check if there is a better option?